pkg/controller/disruption/disruption.go

```diff
@@ -180,15 +188,11 @@ var (
 )
 
 // getPodReplicaSet finds a replicaset which has no matching deployments.
-func (dc *DisruptionController) getPodReplicaSet(pod *v1.Pod) (*controllerAndScale, error) {
-	controllerRef := metav1.GetControllerOf(pod)
-	if controllerRef == nil {
-		return nil, nil
-	}
+func (dc *DisruptionController) getPodReplicaSet(controllerRef *metav1.OwnerReference, namespace string) (*controllerAndScale, error) {
 	if controllerRef.Kind != controllerKindRS.Kind {
```

Comment:
need to also make sure `schema.ParseGroupVersion(controllerRef.APIVersion).Group == ""`

Reply:
Added check for group on all finder functions.